Makefile

```diff
@@ -21,15 +21,33 @@ TEST_CMD=	tests/test_libcperciva.sh
 
 ### Shared code between Tarsnap projects.
 
-all:	cpusupport-config.h
+all:	cpusupport-config.h posix-flags.sh
 	export CFLAGS="$${CFLAGS:-${CFLAGS_DEFAULT}}";	\
-	export "LDADD_POSIX=`export CC=\"${CC}\"; cd ${LIBCPERCIVA_DIR}/POSIX && command -p sh posix-l.sh \"$$PATH\"`";	\
-	export "CFLAGS_POSIX=`export CC=\"${CC}\"; cd ${LIBCPERCIVA_DIR}/POSIX && command -p sh posix-cflags.sh \"$$PATH\"`";	\
+	. ./posix-flags.sh;				\
 	. ./cpusupport-config.h;			\
 	for D in ${PROGS} ${TESTS}; do			\
 		( cd $${D} && ${MAKE} all ) || exit 2;	\
 	done
 
+# To avoid escape characters becoming overwhelming, this generates the _POSIX
+# environment variables first, then writes them to the file.
+posix-flags.sh:
+	if [ -d ${LIBCPERCIVA_DIR}/POSIX/ ];				\
+	then								\
```

Comment:
I would normally put `then` on the same line as `if`, unless there's a reason to do otherwise.